File: admin/backend/src/main/java/com/novel/admin/service/AdminDashboardService.java
package com.novel.admin.service;

import com.novel.admin.dto.DashboardStatsDTO;
import com.novel.admin.mapper.AdminDashboardMapper;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class AdminDashboardService {

    private final AdminDashboardMapper dashboardMapper;

    public DashboardStatsDTO getStats() {
        DashboardStatsDTO stats = new DashboardStatsDTO();
        stats.setTotalUsers(dashboardMapper.countTotalUsers());
        stats.setTotalNovels(dashboardMapper.countTotalNovels());
        stats.setTotalAITasks(dashboardMapper.countTotalAITasks());
        stats.setTotalCost(dashboardMapper.sumTotalCost());
        return stats;
    }

    public Object getUserTrend(int days) {
        return dashboardMapper.getUserTrend(days);
    }

    public Object getAITaskStats() {
        return dashboardMapper.getAITaskStats();
    }

    public Object getRecentTasks(int limit) {
        return dashboardMapper.getRecentTasks(limit);
    }
}

File: admin/frontend/src/services/adminDashboardService.ts
import request from './request'

export const adminDashboardService = {
  getStats: () => {
    return request.get('/dashboard/stats')
  },

  getUserTrend: (days: number = 30) => {
    return request.get('/dashboard/user-trend', { params: { days } })
  },

  getAITaskStats: () => {
    return request.get('/dashboard/ai-task-stats')
  },

  getRecentTasks: (limit: number = 10) => {
    return request.get('/dashboard/recent-tasks', { params: { limit } })
  },
}
